test/utils/admission.go

```diff
@@ -0,0 +1,44 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package utils
+
+import (
+	"reflect"
+	"testing"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/diff"
+	"k8s.io/apiserver/pkg/admission"
+)
+
+// CheckSafeAdmissionReinvocation reinvokes the testFunc, which is expected to call Admit with the proved admission AttributesRecord, and reports a test error
```

Comment:
would probably be good to put this under a `k8s.io/apiserver/pkg/admission/testing` package, for use by downstream admission implementers as well

Reply:
Thanks, I was unsure where the belonged.